tests/client/unit/posts/newPost/newPost.directive.spec.coffee

```diff
@@ -38,23 +38,39 @@ describe 'Directive: newPost', ->
         it 'should clear text', () ->
             expect(element.ctrl.post.text).toBe ''
 
-        it 'should set $stateParams.topic_id to post.topic_di', () ->
+        it 'should set $stateParams.topic_id to post.topic_id', () ->
             expect(element.ctrl.post.topic_id).toBe mockedStateParams.topic_id
 
-    describe 'addPost method', ->
+    describe 'addPost method success', ->
         post =
             text: 'Im Batman'
+        image =
+            name: 'Gotham_Night_Panorama.jpg'
 
         beforeEach (done) ->
             spyOn(element.ctrl, 'clearPost')
-            element.ctrl.addPost(post)
+            element.ctrl.addPost(post, image)
             successCallback()
             done()
 
         it 'should call add method on postsServ with post', () ->
-            expect(mockedPostsServ.add).toHaveBeenCalledWith(post)
+            expect(mockedPostsServ.add).toHaveBeenCalledWith(post, image)
 
         it 'should call clearPost method on controller', () ->
             expect(element.ctrl.clearPost).toHaveBeenCalled()
 
+    describe 'addPost method failure', ->
+        post =
+            text: ''
+        image =
+            name: ''
+
+        beforeEach (done) ->
+            spyOn(element.ctrl, 'clearPost')
+            element.ctrl.addPost(post, image)
+            successCallback()
```

Comment:
successCallback jest to pierwszy argument funkcji then. Then jest methodą od mockedPostsServ.add. Jeżeli metoda add nie jest wywoływana, to then też nie, czyli successCallback tymbardziej nie. Spróbuj przesunąć ten test do góry (jak pierwszy), to powinno wywalić błąd w stylu successCallback is not a function